src/classes/application.js

```diff
@@ -12,22 +13,35 @@ module.exports = class Application extends Base {
     const config = require(path.join(process.cwd(), 'config/environment.js'))(process.env.NODE_ENV)
     const MiddlewareRouter = require(path.join(process.cwd(), 'app/middleware.js'))
     const Router = require(path.join(process.cwd(), 'app/router.js'))
+    const log = new Log()
```

Comment:
I wonder about the wisdom of creating new `Log` instances in various places. You have `new Log()` here in this file, and then again in `src/index.js` below. 

Should Log be a singleton? Should it only ever be initialized as part of this `application.start` method? Should it be directly configurable or indirectly through the `config.js` file? Does it matter that you create multiple instances of Log? Should you not just export `Log` in `src/index.js` instead of `log: new Log()` thus forcing the user to create a `new Log(/* some file-specific configuration like how debug module does */)` in each file?